cuML/src/glm/qn/simple_mat.h

```diff
@@ -245,8 +245,8 @@ inline T nrm2(const SimpleVec<T> &u, T *tmp_dev, cudaStream_t stream = 0) {
 }
 
 template <typename T>
-inline T nrm1(const SimpleVec<T> &u, T *tmp_dev, cudaStream_t stream = 0){
-  MLCommon::LinAlg::rowNorm(tmp_dev, u.data,  u.len, 1, MLCommon::LinAlg::L1Norm,
+inline T nrm1(const SimpleVec<T> &u, T *tmp_dev, cudaStream_t stream = 0) {
+  MLCommon::LinAlg::rowNorm(tmp_dev, u.data, u.len, 1, MLCommon::LinAlg::L1Norm,
                             true, stream, MLCommon::Nop<T>());
   T tmp_host;
   MLCommon::updateHost(&tmp_host, tmp_dev, 1);
```

Comment:
I no this is not related to this PR, but please use `updateHostAsync` with an explicit stream

Reply:
Ok. I suppose then I would need to call `cudaStreamSynchronize` right after?

Reply:
Yes. A stream sync needed right before when it gets consumed.
And ideally speaking, the `tmp_host` should be of type `host_buffer` and that way we could issue this async operation.